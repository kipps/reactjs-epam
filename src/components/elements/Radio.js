import React from 'react';

class Radio extends React.Component {
    state={
        value: ''
    }
    handleChange(event) {
       this.setState({value: event.target.value})
    }
    render() {
        const renderRadioList = (array) => {
            return array.label.map((item) =>
                <label key={item} name={array.name} className='inline ml-8 mr-8'>
                    <span className='inline mr-4'>{item}</span>
                    <input type='radio'
                           onChange={this.handleChange.bind(this)}
                           value={this.state.value}
                           name={array.name}
                    />
                </label>
            );
        };
        return (
            <div className={this.props.className + ' Radio'}>
                {renderRadioList(this.props)}
            </div>
        )
    }
}

export default Radio;