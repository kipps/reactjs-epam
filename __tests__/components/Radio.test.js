import { shallow } from 'enzyme/build';
import { configure } from 'enzyme/build';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16/build';

import Radio from '../../src/components/elements/Radio';

configure({ adapter: new Adapter() });

describe("Radio", () => {
  let searchBy = ['Genre', 'Title'];
  let radioList;

  beforeEach(() => {
    radioList = shallow(<Radio label={searchBy} />);
  });

  it('renders 2 articles', () => {
    expect(radioList.find('div.Radio').length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Radio label={searchBy} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
