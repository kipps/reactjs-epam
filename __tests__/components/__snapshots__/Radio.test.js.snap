// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
<div
  className="undefined Radio"
>
  <label
    className="inline ml-8 mr-8"
  >
    <span
      className="inline mr-4"
    >
      Genre
    </span>
    <input
      type="radio"
    />
  </label>
  <label
    className="inline ml-8 mr-8"
  >
    <span
      className="inline mr-4"
    >
      Title
    </span>
    <input
      type="radio"
    />
  </label>
</div>
`;
